src/languageSetup.ts

```diff
@@ -209,3 +215,15 @@ export function configureLanguage(): void {
         ]
     });
 }
+
+export class KotlinAPI {
+    private buildOutputLocation: string;
+
+    setBuildOutputPath(buildOutputLocation: string) {
+        this.buildOutputLocation = buildOutputLocation;
+    }
+
+    getBuildOutputPath(): string {
+        return this.buildOutputLocation;
+    }
```

Comment:
If we can use ES6-style getters and setters and don't need to perform any special logic, I think exposing the property would be fine for now.

Also do we need a class at all? In a case like this where we only expose a bunch of properties, I think an interface would be sufficient (and even idiomatic, though that's probably a question of style):

```typescript
interface ExtensionApi {
  kotlinApi: KotlinApi;
}

interface KotlinApi {
  buildOutputLocation: string;
}
```

(For consistency, I would use `KotlinApi`/`kotlinApi` or `KotlinAPI`/`kotlinAPI`, whatever you prefer).

Reply:
I think an interface makes sense. With that said, I would prefer not to expose the `KotlinApi` directly and instead keep it private just so we can do internal logic there if we ever need to. This way, we can start having logic in `KotlinApi` without breaking the extension API contract